### DISCUSSION TITLE
Enhancement: add prompt variables/tags that insert tagged content

### DISCUSSION THREAD

**Discussion by insilications** [danny-avila/LibreChat#3503] [Feature Requests & Suggestions]
### What features would you like to see added?

Perhaps this is overkill, but it can be useful for the laziest of people.

### More details

The user defines the variables/tags of the prompt by choosing an opening and closing tag, for example `<cpp>` and `</cpp>`. And the referenced tag can be added inside the prompt text, which in turn would allow any chosen content to have the `<cpp>` tags surround it.

### Which components are impacted by your request?

UI

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Thanks for the feedback. Can you give an example prompt text and the output when used?

**Reply by insilications** [reply]
Thanks for considering this feature. I considered this feature in the context of Claude models being specifically tuned to XML tags when following instructions. Check this: [https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/use-xml-tags](https://docs.anthropic.com/en/docs/build-with-claude/prompt-engineering/use-xml-tags). Even the "Prompt Generator" in the Anthropic dashboard tries to use XML tags when we give it a prompt and ask it to optimize it. Prompt example:

```
You are an expert Bash script analyzer. Your task is:

1. First, carefully analyze and understand the Bash script provided inside the <bash_script> tag:

{{bash_script}}

2. Next, read the explanation of how the analyzed script is supposed to work inside the <explanation> tag:

{{explanation}}

3. Now, follow these specific instructions inside the <instructions> tag to finish your task:

{{instructions}}
```

The user clicks on each detected variable and can define an opening and closing label before saving the prompt:

```
{{bash_script}} is a tag variable.
{{bash_script}} opening tag: <bash_script>
{{bash_script}} closing tag: </bash_script>

{{explanation}} is a tag variable.
{{explanation}} opening tag: <explanation>
{{explanation}} closing tag: </explanation>

{{instructions}} is a tag variable.
{{instructions}} opening tag: <instructions>
{{instructions}} closing tag: </instructions>
```


The output when using the saved prompt and setting their respective contents:

```
You are an expert Bash script analyzer. Your task is:

1. First, carefully analyze and understand the Bash script provided inside the <bash_script> tag:

<bash_script>
"bash_script tag content here"
</bash_script>

2. Next, read the explanation of how the analyzed script is supposed to work inside the <explanation> tag:

<explanation>
"explanation tag content here"
</explanation>

3. Now, follow these specific instructions inside the <instructions> tag to finish your task:

<instructions>
"instructions tag content here"
</instructions>
```

Perhaps this feature is overkill and could add unnecessary complexity to saved prompt system because you have to add a different type of prompt variable to be saved. But maybe it's worth it.
